Telerik.JustMock/Expectations/Local/FunctionExpectation.cs

```diff
@@ -0,0 +1,246 @@
+﻿using System;
+using System.Reflection;
+using Telerik.JustMock.Core;
+using Telerik.JustMock.Expectations.Abstraction.Local;
+using Telerik.JustMock.Expectations.Abstraction.Local.Function;
+
+namespace Telerik.JustMock.Expectations
+{
+	internal sealed class FunctionExpectation : IFunctionExpectation
+	{
+		public ActionExpectation Arrange(object target, string methodName, string localFunctionName, params object[] args)
+		{
+			return ProfilerInterceptor.GuardInternal(() =>
+			{
+				Type[] emptyParamTypes = new Type[] { };
+				return Arrange(target, methodName, emptyParamTypes, localFunctionName, args);
+			});
+		}
+
+		public ActionExpectation Arrange(object target, string methodName, Type[] memberParamTypes, string localFunctionName, params object[] args)
+		{
+			return ProfilerInterceptor.GuardInternal(() =>
+			{
+				MethodInfo method = MockingUtil.GetMethodWithLocalFunction(target, methodName, memberParamTypes);
+				return Arrange(target, method, localFunctionName, args);
+			});
+		}
+		public ActionExpectation Arrange(object target, MethodInfo method, string localFunctionName, params object[] args)
+		{
+			return ProfilerInterceptor.GuardInternal(() =>
+			{
+				MethodInfo localMethod = MockingUtil.GetLocalFunction(target, method, localFunctionName);
+				return Mock.NonPublic.Arrange(target, localMethod, args);
+			});
+		}
+
+		public FuncExpectation<TReturn> Arrange<TReturn>(object target, string methodName, string localFunctionName, params object[] args)
+		{
+			return ProfilerInterceptor.GuardInternal(() =>
+			{
+				Type[] emptyParamTypes = new Type[] { };
+				return Arrange<TReturn>(target, methodName, emptyParamTypes, localFunctionName, args);
+			});
+		}
+
+		public FuncExpectation<TReturn> Arrange<TReturn>(object target, string methodName, Type[] memberParamTypes, string localFunctionName, params object[] args)
+		{
+			return ProfilerInterceptor.GuardInternal(() =>
+			{
+				MethodInfo method = MockingUtil.GetMethodWithLocalFunction(target, methodName, memberParamTypes);
+				return Arrange<TReturn>(target, method, localFunctionName, args);
+			});
+		}
+
+		public FuncExpectation<TReturn> Arrange<TReturn>(object target, MethodInfo method, string localFunctionName, params object[] args)
+		{
+			return ProfilerInterceptor.GuardInternal(() =>
+			{
+				Type type = target.GetType();
+				MethodInfo localMethod = MockingUtil.GetLocalFunction(type, method, localFunctionName);
+
+				return Mock.NonPublic.Arrange<TReturn>(target, localMethod, args);
+			});
+		}
+
+		public object Call(object target, string methodName, string localFunctionName, params object[] args)
+		{
+			return ProfilerInterceptor.GuardInternal(() =>
+			{
+				Type type = target.GetType();
```

Comment:
this method can reuse another Call overload and the body could look like this:

return this.Call(target, methodName, new Type[] { }, localFunctionName, args);